### PR TITLE
Add pylint to `pre-commit`

### LINKED ISSUE #113: Add pylint docstrings to precommit hook
We current check for docstrings on PRs, but it would be nice to have this checked client-side as part of the precommit hook. This would also involve changing the dev requirements and updating the docs.

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
Fixes #113 

Changes proposed by this PR can be summarized as follows :-

- Adds `pylint` to `.pre-commit-config.yaml` using instructions from the official pylint documentation [here](https://pylint.pycqa.org/en/latest/user_guide/pre-commit-integration.html) with appropriate flags from the [**`lint_optional.yml`**](https://github.com/lanl/scico/blob/main/.github/workflows/lint_optional.yml) file.
- Adds `pylint` to `dev_requirements.txt`.

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
@bwohlberg does this change to the docs look okay ?

**Comment by bwohlberg** [comment]
> does this change to the docs look okay?

That looks good to me. I'm fine with the PR as-is, but since @Michael-T-McCann created the original issue, let's wait for his confirmation before merging.

**Review by bwohlberg — APPROVED** [review]

**Review by Michael-T-McCann — APPROVED** [review]

**Comment by bwohlberg** [comment]
Looks good. Thanks for your contribution!

**Pull request merged**
